src/test/java/org/assertj/core/api/object/ObjectAssert_extracting_with_String_and_InstanceOfAssertFactory_Test.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+package org.assertj.core.api.object;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.api.InstanceOfAssertFactories.BIG_DECIMAL;
+import static org.assertj.core.api.InstanceOfAssertFactories.LONG;
+import static org.assertj.core.api.InstanceOfAssertFactories.STRING;
+import static org.assertj.core.error.ShouldNotBeNull.shouldNotBeNull;
+import static org.assertj.core.util.BigDecimalComparator.BIG_DECIMAL_COMPARATOR;
+
+import java.math.BigDecimal;
+import java.util.Comparator;
+
+import org.assertj.core.api.AbstractBigDecimalAssert;
+import org.assertj.core.api.AbstractLongAssert;
+import org.assertj.core.api.AbstractStringAssert;
+import org.assertj.core.api.InstanceOfAssertFactory;
+import org.assertj.core.api.ObjectAssert;
+import org.assertj.core.test.Employee;
+import org.assertj.core.test.Name;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for <code>{@link ObjectAssert#extracting(String, InstanceOfAssertFactory)}</code>.
+ */
+class ObjectAssert_extracting_with_String_and_InstanceOfAssertFactory_Test {
+
+  private Employee luke;
+
+  @BeforeEach
+  void setup() {
+    luke = new Employee(2L, new Name("Luke", "Skywalker"), 26);
+  }
+
+  @Test
+  void should_throw_npe_if_no_assert_factory_is_given() {
+    // WHEN
+    Throwable thrown = catchThrowable(() -> assertThat(luke).extracting("id", null));
+    // THEN
+    then(thrown).isInstanceOf(NullPointerException.class)
+                .hasMessage(shouldNotBeNull("assertFactory").create());
+  }
+
+  @Test
+  void should_allow_type_narrowed_assertions_on_property_extracted_by_name() {
+    // WHEN
+    AbstractLongAssert<?> result = assertThat(luke).extracting("id", LONG);
+    // THEN
+    result.isPositive();
+  }
+
+  @Test
+  void should_allow_narrowed_assertions_on_inner_property_extracted_by_name() {
+    // WHEN
+    AbstractStringAssert<?> result = assertThat(luke).extracting("name.first", STRING);
+    // THEN
+    result.startsWith("Lu");
+  }
+
+  @Test
+  void should_use_property_field_name_as_description_when_extracting_single_property() {
+    // WHEN
+    Throwable thrown = catchThrowable(() -> assertThat(luke).extracting("name.first", STRING)
```

Comment:
use [`expectAssertionError`](https://github.com/joel-costigliola/assertj-core/blob/a58805bbbeb6c5fbbd7b0277d1d66518d7400c65/src/test/java/org/assertj/core/util/AssertionsUtil.java#L24) utility method to directly get the `AssertionError`, same comment for the test below